### PR TITLE
Added Bomb powerup

### PR DESCRIPTION
**Pull request by newspeak1984**

### CODE REVIEW

File: DangoPlop/Assets/Scripts/PowerupMaster.cs [hunk 1]
```diff
@@ -224,5 +225,14 @@ public class PowerupMaster : MonoBehaviour {
 		playerController.rapidFire ();
 	}
 
+	private void explosion(){
+		GameObject[] objects = GameObject.FindGameObjectsWithTag ("Ball");
```
Comment (ktorii) [inline DangoPlop/Assets/Scripts/PowerupMaster.cs]: What does findGameobjectswithtag do exactly to find the array of game objects. Is the array stored in unity's background. Or does it have to loop through every game objects in the game, check if it has the proper tag, and then makes the array from that. 

If it's the latter, then we have to improve the runtime of this explosion function. Explosion should just explode the objects, and not have to search for what to explode too. Runtime=time to search + time to execute split for each ball. 

We want the runtime to be Runtime= time to execute split for each ball.
Ball factory can hold a private variable which is an array which constantly contains the balls in the game.

Comment (newspeak1984) [reply DangoPlop/Assets/Scripts/PowerupMaster.cs]: Say we add an element to the array everytime ball factory makes a ball, when the balls get destroyed, they don't disappear from the array as well do they? If they don't, then there will be a ball object in the array that does not exist in game play

Comment (ktorii) [reply DangoPlop/Assets/Scripts/PowerupMaster.cs]: The game object you destroy from a split should be the same object in the array. If it's not, then we can make it the same by using pointers. Just test and see

Comment (ktorii) [reply DangoPlop/Assets/Scripts/PowerupMaster.cs]: I just looked it up and C# does not support pointers. Which means it's just like Java. If this was java, the ball you destroy will also be destroyed in the array. So the same behavior should be expected for C#

**Pull request closed without merging**